definitions/apply-mapping/apply-mapping.yaml

```diff
@@ -0,0 +1,59 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: apply-mapping
+spec:
+  params:
+    - name: applicationSnapshot
+      type: string
+      description: The ApplicationSnapshot in JSON format to apply the mapping to
+    - name: extraConfigPath
+      type: string
+      description: The path to the config file containing the mapping
+      default: ""
+  workspaces:
+    - name: config
+      description: The workspace where the extra config file containing the mapping resides
+  results:
+    - name: applicationSnapshot
+      description: |
+        The AppicationSnapshot in JSON format with the mapping applied. If there is no mapping key in the
+        extraConfigPath, this result will be the same as the original applicationSnapshot parameter.
+  steps:
+    - name: apply-mapping
+      image: quay.io/hacbs-release/release-utils
+      script: |
+        #!/usr/bin/env sh
+        set -eux
+
+        CONFIG_FILE="$(workspaces.config.path)/$(params.extraConfigPath)"
+        if [ ! -f "${CONFIG_FILE}" ] ; then
+            echo "No valid config file was provided."
+            echo '$(params.applicationSnapshot)' | tee $(results.applicationSnapshot.path)
+            exit 0
+        fi
+        MAPPING_EXISTS=$(yq 'has("mapping")' "${CONFIG_FILE}")
+        if [ "${MAPPING_EXISTS}" != "true" ] ; then
```

Comment:
If the config file contains `---` that command seems to return:
```
---
true
```
Maybe it's better to do `yq '.mapping' ~/validation/config.yaml` and check for not `null`.

Reply:
```[jbieren@jbieren release-bundles]$ yq 'has("mapping")' "review.yaml"
false
[jbieren@jbieren release-bundles]$ cat review.yaml 
---
[jbieren@jbieren release-bundles]$ ```

Reply:
```
[jbieren@jbieren release-bundles]$ yq --version
yq (https://github.com/mikefarah/yq/) version 4.20.1
```